pkg/schema_store/simple_schema_store.go

```diff
@@ -23,21 +23,35 @@ func (store *SimpleSchemaStore) IsInCache(dbName string) bool {
 		return false
 	}
 }
+func (store *SimpleSchemaStore) getFromCache(dbName string, lock bool) (Schema, bool) {
+	if lock {
+		store.RLock()
```

Comment:
how much improvement can we get from usign `RLock`?

Reply:
from the mutex profiler, the delay dropped from 63.12% to 16.87%.